src/commands.cpp

```diff
@@ -3749,6 +3758,14 @@ bool CommandGetUserData::procresult(Result r)
 
         case MAKENAMEID4('*', '!', 'b', 'n'):
             parseUserAttribute(backupNames, versionBackupNames);
+			break;
+
+        case MAKENAMEID6('^', '~', 'j', 's', 'c', 'n'):
```

Comment:
Probably it would be better to use a unique user's attribute, really private (`*` instead of `^`), which are already private and can store more than one item. Using a protected user's attribute to store something private looks a weird to me.

I'll consider if changes in that direction can simplify the code, and proceed accordingly.

Reply:
The main reason I stored it using the protected, instead of private, scope is that it avoided me storing something that's already effectively encrypted in a TLV container.

Before these attributes are used, they are ECB decrypted using the master key.

Reply:
The records in the TLV container don't need to be encrypted, they can be stored in plain-text (name and key bytes).
The TLV container is serialized and encrypted before sending to API (and decrypted when they are received).

I think I didn't get your point about storing something that is encrypted :thinking:

Reply:
The user attributes themselves are not encrypted: They are simply random bytes stored as Base64.

That's okay though as the values we are storing are not the actual values we'll be using for say, the authentication or encryption keys.

The value we use for those tasks are the value of the attributes, ECB decrypted using the master key.

In this sense, the value of the user attributes are effectively encrypted: They're no use at all until they are decrypted using the master key.

Reply:
Put differently: We're simply decrypting random bytes using the master key.

The result of the decrypt is the true value of the user attributes.

Reply:
Yes, that's clear to me. The key point is that you are artificially encrypting the actual value of the user attribute and then storing it in the protected attr. If you had used a private user attr, you won't need to apply such encryption on the value, since the attr is encrypted as a whole before sending to API (and when received from).
So for the key bytes, in example:
```
^~jsck = encrypt(<key_bytes>, master_key)
```
while using a private attr:
```
*jsck = encrypt(<serialized_TLV_container>, master_key)
```
So you can store both records, `name` and `key`, in the same TLV container, and the SDK will automatically deal with encryption, in order to keep the value private for the user (the API cannot decrypt them).